cds_ils/importer/documents/api.py

```diff
@@ -0,0 +1,128 @@
+import click
+from elasticsearch_dsl import Q
+from elasticsearch_dsl.query import Query, Bool
+from invenio_app_ils.documents.api import Document
+from invenio_app_ils.documents.search import DocumentSearch
+
+from cds_ils.importer.errors import DocumentImportError
+
+
+def check_search_results(result, search_term, search_term_name,
+                         raise_on_empty=False, raise_on_many=True):
+    """Check number of search results"""
+    hits_total = result.hits.total.value
+
+    if hits_total < 1:
+
+        click.secho(
+            "no document found with {} {}".format(search_term_name,
+                                                  search_term),
+            fg="red",
+        )
+        if raise_on_empty:
+            raise DocumentImportError(
+                "no document found with {} {}".format(search_term_name,
+                                                      search_term)
+            )
+    elif hits_total > 1:
+        click.secho(
+            "found more than one document with {} {}".format(search_term_name,
+                                                             search_term),
+            fg="red",
+        )
+        if raise_on_many:
+            raise DocumentImportError(
+                "found more than one document with {} {}".format(
+                    search_term_name, search_term)
+            )
+    return hits_total
+
+
+def search_documents_by_isbn(isbn):
+    """Find document by ISBN."""
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", identifiers__scheme="ISBN"),
+            Q("match", identifiers__value=isbn),
+        ],
+    )
+    return search.execute()
+
+
+def search_documents_by_doi(doi):
+    """Find document by ISBN."""
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", identifiers__scheme="DOI"),
+            Q("match", identifiers__value=doi),
+        ],
+    )
+    return search.execute()
+
+
+def get_document_by_exact_title_authors(title, authors):
+    """Find document by title and authors."""
+
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", title=title),
```

Comment:
I think that for **exact** match, you need to use `term` queries:
```suggestion
            Q("term", title=title),
```
but I might be wrong, to be checked (see [here](https://stackoverflow.com/questions/40124317/what-is-the-difference-between-a-term-query-and-a-match-one). With `match`, you might get results ignoring upper/lower cases, but also on other cases because the match will be done on the analyzed terms.
Or maybe you want this, but then the function name should be different...
```